From dcd06b6208d0f8dbe45471e61a9dab77c53b5e08 Mon Sep 17 00:00:00 2001
From: Valentin Caron <valentin.caron@foss.st.com>
Date: Mon, 8 Dec 2025 14:53:22 +0100
Subject: [PATCH 940/954] fix(stm32mp1): increase STM32MP_BL2_SIZE

Due to compile issues for stm32mp157c-ed1 emmc, increase STM32MP_BL2_SIZE
and decrease STM32MP_BL2_RO_SIZE of 4kB.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git fd9d8b661c3d1d799d938b2eb9a85e005dfde1f9 fix(stm32mp1): increase STM32MP_BL2_SIZE]
Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: I3754b5b7fb197007312694f9fdafa21a74e290a9
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/516019
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 plat/st/stm32mp1/stm32mp1_fip_def.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/plat/st/stm32mp1/stm32mp1_fip_def.h b/plat/st/stm32mp1/stm32mp1_fip_def.h
index 8ad1ed9c76..34432e1a62 100644
--- a/plat/st/stm32mp1/stm32mp1_fip_def.h
+++ b/plat/st/stm32mp1/stm32mp1_fip_def.h
@@ -49,8 +49,8 @@
 #endif
 #endif /* STM32MP13 */
 #if STM32MP15
-#define STM32MP_BL2_RO_SIZE		U(0x00011000)	/* 68 KB */
-#define STM32MP_BL2_SIZE		U(0x00016000)	/* 88 KB for BL2 */
+#define STM32MP_BL2_RO_SIZE		U(0x00010000)	/* 64 KB */
+#define STM32MP_BL2_SIZE		U(0x00017000)	/* 92 KB for BL2 */
 #endif /* STM32MP15 */
 #endif /* TRUSTED_BOARD_BOOT && !STM32MP_USE_EXTERNAL_HEAP */
 
-- 
2.53.0

